### ISSUE TITLE
Rancher 2.2.1 kube-apiserver in loop missing stat /etc/kubernetes/kube-api-authn-webhook.yaml: no such file or directory

### ISSUE THREAD

**Issue by panho66** [rancher/rancher#19438]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Create Rancher HA, using Rke and install Rancher 
Create a new cluster
Follow dialog cut and paste command in target hosts:

```
docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.2.1 --server https://URL --token 9nbnhpbzx7tcbfkvjwjc7292kvb2h5hm5t8h2hvc5szckqrnrdbzjm --ca-checksum b93b8de5dc981ffe4c3774c960ebdd7ce796d5a4a733f010d6e874f6b9ed7718 --etcd --controlplane --worker
```

**Result:**
**Rancher GUI**
This cluster is currently Provisioning; areas that interact directly with it will not be available until the API is ready.

[controlPlane] Failed to bring up Control Plane: [Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [10.139.157.241]: Get https://localhost:6443/healthz: dial tcp [::1]:6443: connect: connection refused, log: invalid authentication config: stat /etc/kubernetes/kube-api-authn-webhook.yaml: no such file or directory]

**some container starts up but kube-apiserver in crash loop**
`invalid authentication config: stat /etc/kubernetes/kube-api-authn-webhook.yaml: no such file or directory`

**docker ps output**
```
CONTAINER ID        IMAGE                                  COMMAND                  CREATED             STATUS                          PORTS               NAMES
4126f9970e51        rancher/hyperkube:v1.13.5-rancher1     "/opt/rke-tools/en..."   7 minutes ago       Restarting (1) 43 seconds ago                       kube-apiserver
d64fe1f72331        rancher/coreos-etcd:v3.2.24-rancher1   "/usr/local/bin/et..."   8 minutes ago       Up 8 minutes                                        etcd
00191ec9006e        rancher/rancher-agent:v2.2.1           "run.sh --server h..."   9 minutes ago       Up 9 minutes                                        jolly_engelbart
a612402b5cc8        rancher/rancher-agent:v2.2.1           "run.sh -- share-r..."   9 minutes ago       Up 9 minutes                                        share-mnt
8446c92f1c67        rancher/rancher-agent:v2.2.1           "run.sh --server h..."   9 minutes ago       Up 9 minutes                                        kind_archimedes
```

**Other details that may be helpful:**
kube-apiserver log file
https://raw.githubusercontent.com/panho66/logs/master/api.log
etcd log file
https://raw.githubusercontent.com/panho66/logs/master/etcd.log
jolly_engelbart log file ( rancher/rancher-agent:v2.2.1)
https://raw.githubusercontent.com/panho66/logs/master/jolly_engelbart.log
share-mnt log file (rancher/rancher-agent:v2.2.1) 
https://raw.githubusercontent.com/panho66/logs/master/share-mnt.log

**Manually create file /etc/kubernetes/kube-api-authn-webhook.yaml**
```
apiVersion: v1
kind: Config
clusters:
- name: Default
  cluster:
    insecure-skip-tls-verify: true
    server: http://127.0.0.1:6440/v1/authenticate
users:
- name: Default
  user:
    insecure-skip-tls-verify: true
current-context: webhook
contexts:
- name: webhook
  context:
    user: Default
    cluster: Default
```

**Node successfully added** 

kube-apiserver log after create file manually
https://raw.githubusercontent.com/panho66/logs/master/kube-api-ok.log

**Environment information**
Rancher version 2.2.1
Red Hat Enterprise Linux Server release 7.6 (Maipo)
```
Client:
 Version:         1.13.1
 API version:     1.26
 Package version: docker-1.13.1-91.git07f3374.el7.x86_64
 Go version:      go1.10.3
 Git commit:      07f3374/1.13.1
 Built:           Fri Feb  8 20:24:43 2019
 OS/Arch:         linux/amd64

Server:
 Version:         1.13.1
 API version:     1.26 (minimum version 1.12)
 Package version: docker-1.13.1-91.git07f3374.el7.x86_64
 Go version:      go1.10.3
 Git commit:      07f3374/1.13.1
 Built:           Fri Feb  8 20:24:43 2019
 OS/Arch:         linux/amd64
 Experimental:    false
```

**Comment by panho66** [comment]
is this workaround OK? Is the cluster still health ?
got a lot of this message:
I0408 10:47:21.439477       1 controller.go:105] OpenAPI AggregationController: Processing item v1beta1.metrics.k8s.io
I0408 10:47:21.445325       1 controller.go:116] OpenAPI AggregationController: action for item v1beta1.metrics.k8s.io: Requeue.
I0408 10:48:21.445709       1 controller.go:105] OpenAPI AggregationController: Processing item v1beta1.metrics.k8s.io
I0408 10:48:21.451245       1 controller.go:116] OpenAPI AggregationController: action for item v1beta1.metrics.k8s.io: Requeue.
I0408 10:49:21.451554       1 controller.go:105] OpenAPI AggregationController: Processing item v1beta1.metrics.k8s.io
I0408 10:49:21.466012       1 controller.go:116] OpenAPI AggregationController: action for item v1beta1.metrics.k8s.io: Requeue.

**Comment by superseb** [comment]
Can you supply `df -h` and `mount` from the host? Is this happening on every RHEL 7.6 host you are trying to add?

**Comment by superseb** [comment]
Also the output of `sestatus` and the contents of `/var/log/audit/audit.log` might be useful.

**Comment by panho66** [comment]
I have Rancher local cluster built with RKE 3 nodes
Add cluster1 3 nodes ( non-prod ) no problem
Add cluster2 3 nodes (non-prod) no problem
**Add cluster3 2 nodes (prd ) same problem** 

mount log
https://raw.githubusercontent.com/panho66/logs/master/mount.log

audit log
https://raw.githubusercontent.com/panho66/logs/master/audit.log
audit log.1
https://raw.githubusercontent.com/panho66/logs/master/audit.log.1
https://raw.githubusercontent.com/panho66/logs/master/audit.log.2
https://raw.githubusercontent.com/panho66/logs/master/audit.log.3

/home/dockeradm/logs # sestatus
SELinux status:                 disabled

/home/dockeradm/logs # df -h
Filesystem                               Type      Size  Used Avail Use% Mounted on
/dev/mapper/rootvg-lv_root               ext4      9.8G  1.8G  7.5G  20% /
devtmpfs                                 devtmpfs  8.8G     0  8.8G   0% /dev
tmpfs                                    tmpfs     8.8G     0  8.8G   0% /dev/shm
tmpfs                                    tmpfs     8.8G  173M  8.6G   2% /run
tmpfs                                    tmpfs     8.8G     0  8.8G   0% /sys/fs/cgroup
/dev/sda1                                ext4      488M  338M  115M  75% /boot
/dev/mapper/rootvg-lv_var                ext4      7.8G  1.2G  6.2G  16% /var
/dev/mapper/datavg-lv_vartsm             ext4      2.0G   16M  1.8G   1% /var/tsm
/dev/mapper/rootvg-lv_home               ext4      3.9G   56M  3.6G   2% /home
/dev/mapper/rootvg-lv_opt                ext4      9.8G  867M  8.4G  10% /opt
/dev/mapper/datavg-lv_optsplunkforwarder ext4      3.9G  601M  3.1G  17% /opt/splunkforwarder
/dev/mapper/rootvg-lv_tmp                ext4      4.9G   40M  4.6G   1% /tmp
/dev/mapper/datavg-lv_iag                ext4      197G   59M  189G   1% /iag
/dev/mapper/datavg-lv_libdocker          ext4      110G  104M  104G   1% /var/lib/docker
/dev/mapper/datavg-lv_etcd               ext4       11G  167M   10G   2% /var/lib/etcd
/dev/mapper/datavg-lv_rke                ext4       11G   41M   11G   1% /opt/rke
/dev/mapper/datavg-lv_pods               ext4       11G   41M   11G   1% /var/log/pods
/dev/mapper/datavg-lv_containers         ext4       11G   41M   11G   1% /var/log/containers
/dev/mapper/datavg-lv_rancher            ext4       11G   41M   11G   1% /var/lib/rancher
/dev/mapper/datavg-lv_kubelet            ext4      3.4G   14M  3.2G   1% /var/lib/kubelet
tmpfs                                    tmpfs     1.8G     0  1.8G   0% /run/user/99
tmpfs                                    tmpfs     1.8G     0  1.8G   0% /run/user/39103
tmpfs                                    tmpfs     1.8G     0  1.8G   0% /run/user/0

**Comment by superseb** [comment]
Can you share output of `docker info` and the output of this command:

```
time docker run --rm -v /etc/kubernetes:/etc/kubernetes:z rancher/rke-tools:v0.1.27 sh -c 't=$(mktemp); mv $t /etc/kubernetes/$(basename $t)'
```

**Comment by greenu** [comment]
I have same issue and you've posted this just while I had a chat in slack with @superseb on that.

**Comment by superseb** [comment]
Current theory:

The file operation inside the file-deployer container takes too long, and the container gets removed before it can complete in the next step in the provisioning process. This logging is with a sleep 10 in the file operation one-liner, and the container get started and removed within 1 second:

```
2019/04/08 15:02:52 [INFO] cluster [c-2nrn8] provisioning: [file-deploy] Deploying file '/etc/kubernetes/kube-api-authn-webhook.yaml' to node [x]
2019/04/08 15:02:52 [INFO] cluster [c-2nrn8] provisioning: [file-deploy] Successfully started [file-deployer] container on host [x]
2019/04/08 15:02:53 [INFO] cluster [c-2nrn8] provisioning: [remove/file-deployer] Successfully removed container on host [x]
2019/04/08 15:02:53 [INFO] cluster [c-2nrn8] provisioning: [/etc/kubernetes/cloud-config] Successfully deployed authentication webhook config Cluster nodes
```

**Comment by panho66** [comment]
time docker run --rm -v /etc/kubernetes:/etc/kubernetes:z rancher/rke-tools:v0.1.27 sh -c 't=$(mktemp); mv $t /etc/kubernetes/$(basename $t)'

real    0m2.133s
user    0m0.047s
sys     0m0.048s


/home/dockeradm # docker info
Containers: 21
 Running: 15
 Paused: 0
 Stopped: 6
Images: 13
Server Version: 1.13.1
Storage Driver: devicemapper
 Pool Name: datavg-docker_pool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 4.098 GB
 Data Space Total: 107.4 GB
 Data Space Available: 103.3 GB
 Metadata Space Used: 18.6 MB
 Metadata Space Total: 5.365 GB
 Metadata Space Available: 5.346 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.149-RHEL7 (2018-07-20)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version:  (expected: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1)
runc version: 290a33602b16ff2d1cc5339bc0297f0e094462ce (expected: 9df8b306d01f59d3a8029be411de015b7304dd8f)
init version: N/A (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-957.5.1.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.6 (Maipo)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 3
CPUs: 4
Total Memory: 17.46 GiB
Name: ulpdkr159
ID: KI2E:5NA4:WZQL:EQ35:KYEJ:LTVI:JNIF:7BRX:6GWF:SWGX:K53V:PDWG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Http Proxy: http://proxy:8080
Https Proxy: http://proxy:8080
No Proxy: 127.0.0.1,localhost,10.139.157.159,ulpdkr159
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Registries: docker.io (secure)

**Comment by panho66** [comment]
@superseb does my workaround ok by manually create file ?

**Comment by panho66** [comment]
@superseb I have removed log files due to security policy in my company. If you need it, please let me know

**Comment by panho66** [comment]
https://github.com/rancher/rancher/issues/19572

**Comment by alena1108** [comment]
@superseb I know this bug wasn't easy to repro, and you've used some script to emulate the scenario. Can you share it with QA?

**Comment by superseb** [comment]
See https://github.com/rancher/rancher/issues/19515#issuecomment-486989323, only difference for 2.3 is the `rke-tools` version which is currently `v0.1.28` for this version.

**Comment by jira-sync-svc** [comment]
➤ Jack Luo commented:

Following the steps from the above comments, the bug is reproduced in Rancher v2.2.2, and the big fix is validated in Rancher master c3dd4e2d
